vllm/v1/core/scheduler.py

```diff
@@ -223,6 +294,50 @@ def _make_running_request_data(
             self.running_reqs_data[request.request_id] = req_data
         return req_data
 
+    def _schedule_encoder_inputs(
+        self,
+        request: Request,
+        num_computed_tokens: int,
+        num_new_tokens: int,
+        encoder_budget: int,
+    ) -> Tuple[List[int], int]:
+        if not request.has_encoder_inputs():
+            return [], num_new_tokens
+
+        encoder_inputs_to_schedule: List[int] = []
+        mm_positions = request.mm_positions
+        assert mm_positions is not None
+        assert len(mm_positions) > 0
+        for i, pos_info in enumerate(mm_positions):
```

Comment:
I wonder if it makes sense to evict `mm_position` that has been fully processed because it seems that we will be unnecessarily iterating over them a lot (here and later in `_gather_encoder_outputs`)